ui/src/main/webapp/index.js

```diff
@@ -29,13 +31,15 @@ if (process.env.NODE_ENV !== 'production') {
     // use <App /> here rather than require() a <NextApp />.
     try {
       const routes = require('./app').routes
-      render(
+      instance = render(
         <AppContainer errorReporter={({ error }) => { throw error }}>
           <Router history={hashHistory} routes={routes} />
         </AppContainer>,
         document.getElementById('root'))
     } catch (e) {}
   })
+
+  window.forceUpdateThemeing = () => setTimeout(() => deepForceUpdate(instance), 0)
```

Comment:
The muiTheme object is passed using react's context api which does not reliably update child components, so we force a rerender of all components on a theme color change. We thought it was appropriate to use in this case since all components are (or should be) affected by a theming change. The reason we specifically named it `forceUpdateThemeing` is to prevent anyone from using this unless it's specifically a theme-related update. Nothing else should need to use this.